libgadget/blackhole.c

```diff
@@ -471,6 +634,148 @@ blackhole(const ActiveParticles * act, ForceTree * tree, FILE * FdBlackHoles, FI
     walltime_measure("/BH");
 }
 
+
+/*************************************************************************************/
+/* DF routines */
+static void
+blackhole_dynfric_postprocess(int n, TreeWalk * tw){   
+
+    int PI = P[n].PI;
+
+    /***********************************************************************************/
+    /* Compute/add accel. when DF turned on */
+    /* averaged value for colomb logarithm and integral over the distribution function */
+    /* acc_friction = -4*pi*G^2 * Mbh * log(lambda) * rho * f_of_x * bhvel / |bhvel^3| */
+    /*       f_of_x = [erf(x) - 2*x*exp(-x^2)/sqrt(pi)]                                */
+    /*       lambda = b_max * v^2 / G / (M+m)                                          */
+    /*        b_max = Size of system (e.g. Rvir)                                       */
+    /*            v = Relative velocity of BH with respect to the environment          */
+    /*            M = Mass of BH                                                       */
+    /*            m = individual mass elements composing the large system (e.g. m<<M)  */
+    /*            x = v/sqrt(2)/sigma                                                  */
+    /*        sigma = width of the max. distr. of the host system                      */
+    /*                (e.g. sigma = v_disp / 3                                         */
+
+    if(blackhole_params.BH_DynFrictionMethod > 0 && BH_GET_PRIV(tw)->BH_SurroundingDensity[PI] > 0){ 
+
+        double bhvel;
+        double lambda, x, f_of_x;
+        const double a_erf = 8 * (M_PI - 3) / (3 * M_PI * (4. - M_PI));
+
+
+        /* normalize velocity/dispersion */
+        BH_GET_PRIV(tw)->BH_SurroundingRmsVel[PI] /= BH_GET_PRIV(tw)->BH_SurroundingDensity[PI];
+        BH_GET_PRIV(tw)->BH_SurroundingRmsVel[PI] = sqrt(BH_GET_PRIV(tw)->BH_SurroundingRmsVel[PI]);
+        for(int k = 0; k < 3; k++)
```

Comment:
We avoid declaring variables in the loop body because (I think) one of the compilers on BlueTides doesn't support it properly. Could you move the variable declaration of of the loop? Just use j for all the loops in this clause.